src/v/datalake/datalake_manager.cc

```diff
@@ -282,35 +281,33 @@ ss::future<> datalake_manager::start() {
     co_await _backlog_controller->start();
 
     /*
-     * Start the global disk space usage monitor loop
+     * Initialize the reservation pool and start the global disk space usage
+     * monitor loop
      */
     const ss::shard_id disk_space_monitor_core = 0;
     if (ss::this_shard_id() == disk_space_monitor_core) {
+        _disk_space_available.signal(
+          config::shard_local_cfg().datalake_scratch_space_size_bytes());
         ssx::spawn_with_gate(_gate, [this] { return disk_space_monitor(); });
     }
+}
+
+size_t datalake_manager::disk_space_soft_limit() {
+    return config::shard_local_cfg()
+      .datalake_scratch_space_soft_limit_size_bytes();
+}
 
-    _disk_usage_interval.watch([this] { _disk_space_monitor_sem.signal(); });
+bool datalake_manager::disk_space_soft_limit_exceeded() {
+    return _disk_space_available.current() < disk_space_soft_limit();
```

Comment:
Either this code is incorrect or the naming is confusing. Is soft limit a "limit" i.e. with 1GiB value we breach it when we go above 1GiB usage? Or, soft limit is actually "free space" soft "target"? The naming implies the former whereas the code implies the later. 🤔

Reply:
Good eye. You probably missed it since it wasn't broken out into a separate commit, but this is fixed in a later commit: https://github.com/redpanda-data/redpanda/pull/25766/commits/551f0d037345614fa10690e7284ad6b3402c05d2#r2047124862. Does that address your concern?

Reply:
Can you update the bits from that commit into this one? It's kind of hard as a reviewer to reason about this commit alone